### PR TITLE
Expose payout endpoints

### PR DESCRIPTION
**Pull request by OdunlamiZO**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -60,6 +61,16 @@ type RemoteCalls interface {
 	GenerateBankAccount(ctx context.Context, request model.BankAccountRequest) (model.BankAccountResponse, error)
 	GetBankAccount(ctx context.Context, customerID uuid.UUID) (model.BankAccountResponse, error)
 
+	// Payout APIs
+	GetPayoutByID(ctx context.Context, payoutID string) (model.PayoutResponse, error)
+	InitiateDirectBulkPayout(ctx context.Context, request model.InitiateBulkPayoutRequest) (model.PayoutDetails, error)
+	InitiatePayout(ctx context.Context, currency, payoutType, beneficiaryType, remarks string, document *os.File) (model.PayoutDetails, error)
+	GetAllPayouts(ctx context.Context, status, search string, dateBetween model.DateBetween, page model.Page) (model.AllPayoutsResponse, error)
+	CancelPayout(ctx context.Context, request model.CancelPayoutRequest) (bool, error)
+	UpdatePayoutAccount(ctx context.Context, payoutID string, request model.TransferBeneficiaryDetails) (bool, error)
```
Comment (odetolakehinde) [inline api/api.go:70]: these should just return error.

File: api/payout.go [hunk 2]
```diff
@@ -3,82 +3,160 @@ package api
...
+		if page.Size != nil {
+			params["size"] = strconv.Itoa(*page.Size)
+		}
+		if page.SortBy != "" {
+			params["sort_by"] = page.SortBy
+		}
+		if page.SortDirectionDesc != nil {
+			params["sort_direction_desc"] = strconv.FormatBool(*page.SortDirectionDesc)
+		}
+	}
```
Comment (odetolakehinde) [inline api/payout.go:97]: create an helper function for this. the `dateBetween` also.

File: api/api.go (hunk 1 above)
Comment (OdunlamiZO) [reply api/api.go:70]: Alright

File: model/model.go [hunk 3]
```diff
@@ -677,4 +698,63 @@ type (
...
+
+	// CancelPayoutRequest request schema for cancel payout
+	CancelPayoutRequest struct {
+		BulkPayoutID string `json:"payout_id"`
+		Reason       string `json:"reason"`
+	}
+
+	// BulkPayoutConfig schema for payout config
+	BulkPayoutConfig struct {
+		Provider                 string  `json:"provider"`
```
Comment (odetolakehinde) [inline model/model.go:710]: create a `payout.go` file in the model package to house all payout related models.

File: api/util.go [hunk 4]
```diff
@@ -114,3 +125,39 @@ func mapstruct(data, v interface{}) error {
...
+		}
+
+		str, ok := data.(string)
+		if !ok {
+			return data, nil
+		}
+
+		return time.Parse(time.RFC3339, str)
+	}
+}
```
Comment (odetolakehinde) [inline api/util.go:163]: wanna write tests for these?

**Review by odetolakehinde — CHANGES_REQUESTED** [review]
some comments sir.

File: model/model.go [hunk 5]
```diff
@@ -654,11 +665,21 @@ type (
...
-		Currency string                  `json:"currency"`
-		Remarks  string                  `json:"remarks"`
-		Accounts []InitiatePayoutAccount `json:"accounts"`
+	// InitiateBulkPayoutRequest schema
+	InitiateBulkPayoutRequest struct {
+		Currency        string                       `json:"currency"`
+		Remarks         string                       `json:"remarks"`
+		Accounts        []BulkPayoutRecipientAccount `json:"accounts"`
+		BeneficiaryType PayoutType                   `json:"beneficiary_type"`
+		BeneficiaryID   *string                      `json:"beneficiary_id"`
```
Comment (odetolataiwo) [inline model/model.go:674]: You don't need to add `omitempty` to this?

File: model/model.go [hunk 6]
```diff
@@ -654,11 +665,21 @@ type (
...
+		Amount          *float64                     `json:"amount"`
+	}
+
+	// BulkPayoutRecipientAccount schema
+	BulkPayoutRecipientAccount struct {
+		Amount      float64                    `json:"amount"`
+		Destination TransferBeneficiaryDetails `json:"destination"`
+		Remarks     string                     `json:"remarks"`
```
Comment (odetolataiwo) [inline model/model.go:682]: Is this a compulsory field?

File: api/payout.go (hunk 2 above)
Comment (OdunlamiZO) [reply api/payout.go:97]: Alright sir

File: model/model.go (hunk 5 above)
Comment (OdunlamiZO) [reply model/model.go:674]: I will add them accordingly

File: model/model.go (hunk 6 above)
Comment (OdunlamiZO) [reply model/model.go:682]: I don't get this sir?

File: model/model.go (hunk 3 above)
Comment (OdunlamiZO) [reply model/model.go]: Alright sir

File: api/util.go (hunk 4 above)
Comment (OdunlamiZO) [reply api/util.go]: I will do that

**Review by odetolataiwo — APPROVED** [review]

**Review by oluwafeyipitan — APPROVED** [review]

**Review by odetolakehinde — APPROVED** [review]

**Pull request merged**